src/main/java/de/komoot/photon/nominatim/NominatimUpdater.java

```diff
@@ -39,25 +39,28 @@ public void update() {
                 for (UpdateRow place : getIndexSectorPlaces(rank, (Integer) sector.get("geometry_sector"))) {
 
                     template.update("update placex set indexed_status = 0 where place_id = ?", place.getPlaceId());
-                    final PhotonDoc updatedDoc = exporter.getByPlaceId(place.getPlaceId());
-
-                    switch (place.getIndexdStatus()) {
-                        case 1:
-                            if (updatedDoc.isUsefulForIndex())
-                                updater.create(updatedDoc);
-                            break;
-                        case 2:
-                            if (!updatedDoc.isUsefulForIndex())
+                    final List<PhotonDoc> updatedDocs = exporter.getByPlaceId(place.getPlaceId());
+
+                    for (PhotonDoc updatedDoc : updatedDocs) {
+                        switch (place.getIndexdStatus()) {
+                            case 1:
+                                if (updatedDoc.isUsefulForIndex())
+                                    updater.create(updatedDoc);
+                                break;
+                            case 2:
+                                if (!updatedDoc.isUsefulForIndex()) {
+                                    updater.delete(place.getPlaceId());
+                                    break;
+                                }
+                                updater.updateOrCreate(updatedDoc);
+                                break;
+                            case 100:
                                 updater.delete(place.getPlaceId());
-
-                            updater.updateOrCreate(updatedDoc);
-                            break;
-                        case 100:
-                            updater.delete(place.getPlaceId());
-                            break;
-                        default:
-                            LOGGER.error(String.format("Unknown index status %d", place.getIndexdStatus()));
-                            break;
+                                break;
+                            default:
+                                LOGGER.error(String.format("Unknown index status %d", place.getIndexdStatus()));
+                                break;
+                        }
```

Comment:
Looking at this again, the different cases need a bit more consideration.

For 'case 100' (object was deleted), export.getByPlaceId() shouldn't even be called because we already know that the placeID won't exist anymore. (That was wrong before as well but is worth fixing while changing this code.)

More importantly, for the 'case 2', the code does not catch when a housenumber gets deleted from the batch of housenumbers for that placeID because the code only updates the ones that exist after the update. The most simple solution here is to always call a delete with a subsequent create for case 2. I don't know what the performance impact of such a solution would be.

In any case, the delete() in 'case 2' should be moved out of the loop. `isUsefulForIndex()` evaluates always to the same value for the elements of `updatesDocs` and calling delete() multiple times for the same placeID is inefficient, if not harmful.

Reply:
I've fixed the case '100' issue. 

I'm not really sure how to fix the other case '2' issue. We're already running some months with this code, so it doesn't seems to be harmful. This PR looks quite important to me, because the update process is really broken now, if anyone now how to fix it, it would be great.

Reply:
I think @lonvia wanted something along the lines of
```

                   final List<PhotonDoc> updatedDocs = exporter.getByPlaceId(place.getPlaceId());
                    if (place.getIndexdStatus() == 2 && updatedDocs.size() > 1) {
                        updater.delete(placeId);
                    }
                    for (PhotonDoc updatedDoc : updatedDocs) {
                        switch (place.getIndexdStatus()) {
                        case 1:
                            if (updatedDoc.isUsefulForIndex()) {
                                updater.create(updatedDoc);
                            }
                            break;
                        case 2:
                            if (updatedDoc.isUsefulForIndex()) {
                                updater.updateOrCreate(updatedDoc);
                            }
                            break;
```

While not elegant it should at least be correct, what the current code isn't.

Reply:
@hendrikmoree I think @lonvia was thinking about something along

```
                    final List<PhotonDoc> updatedDocs = exporter.getByPlaceId(place.getPlaceId());
                    if (indexdStatus == 2 && updatedDocs.size() > 1) {
                        updater.delete(placeId);
                    }
                    boolean wasUseful = false;
                    for (PhotonDoc updatedDoc : updatedDocs) {
                        switch (indexdStatus) {
                        case 1:
                            if (updatedDoc.isUsefulForIndex()) {
                                updater.create(updatedDoc);
                            }
                            break;
                        case 2:
                            if (updatedDoc.isUsefulForIndex()) {
                                updater.updateOrCreate(updatedDoc);
                                wasUseful = true;
                            }
                            break;
                        default:
                            LOGGER.error(String.format("Unknown index status %d", indexdStatus));
                            break;
                        }
                    }
                    if (indexdStatus == 2 && !wasUseful) {
                        updater.delete(placeId);
                    }
```
This is better, but still has a couple of cases that are not covered, for example that the object now only has 1 house number and used to have multiple.